subsys/settings/src/settings_shell.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <settings/settings.h>
+#include <shell/shell.h>
+#include <sys/util.h>
+#include <toolchain.h>
+
+#include <stdint.h>
+#include <stddef.h>
+
+/*
+ * Defines the maximum size of a setting that can be read entirely.
+ * In case one attempts to read a value longer than that, the output will be truncated.
+ */
+#define MAX_READ_VALUE_LENGTH 128
```

Comment:
I'm not familiar with how big settings are. But if there is a reasonable chance 128 could be too small some time in the future this should be a Kconfig.

Reply:
How does this correspond to https://github.com/zephyrproject-rtos/zephyr/blob/main/include/settings/settings.h#L35?

Reply:
Right, I overlooked that setting.

Reply:
Although I can see that  `SETTINGS_MAX_VAL_LEN` is hardly used in the code. No settings backends respect that. But still it's better than creating yet another constant.